### ISSUE TITLE
docker image is failing

### ISSUE THREAD

**Issue by bourgeoa** [nodeSolidServer/node-solid-server#1798]
@angelo
since 5.8.0

**Comment by melvincarvalho** [comment]
> Could it be the use of `node --experimental-require-module`

Yes—we shouldn’t be running with this experimental flag. It was deprecated in Node 20 and removed entirely in Node 21. Using experimental flags makes the Docker build fragile and incompatible with modern Node versions. Docker is important to NSS, and this breaks it.

This feels like a breaking change that was merged too early. I’d strongly favour reverting it, or at the very least, moving it to a separate branch until it’s properly supported and integrated.

We’re already on node:lts (Node 20.x), so it’s time to drop or isolate this.

**Comment by melvincarvalho** [comment]
I suggest reverting to 5.7 and making a new experimental branch for experimental features that can be checked out and run on an instance with instructions on how to do that.

**Comment by bourgeoa** [comment]
`node --experimental-require-module` may have been the issue for docker.
It shall not be anymore an issue because `require-module` has been backported in node 20.19.0 and node 22.14.0

Branch `fix/issue#1798` has been created to remove the need for `experimental-require-module` parameter.
Locally `tests` passes without any issue.
But CI is failing see #1817 

If you have any hint I need help.
For now we cannot make any patch to NSS with CI

**Comment by melvincarvalho** [comment]
The sarcastic and dismissive tone here — e.g. *“let’s play logical fallacies again”* and *“failure to understand”* — is not okay. It crosses into personal attack and violates the expectations of respectful collaboration set out in the W3C Code of Conduct. Let’s please keep this space professional and welcoming to all contributors.

On the technical issue:

- The Docker image has been broken since 5.8.0 — that’s a regression in real-world usage.  
- It was caused by reliance on a deprecated, experimental flag removed in Node 21+.  
- Whether or not it’s technically backported doesn’t change the fact that it broke builds.

**IMHO, we should revert to 5.7.x as the stable release**, and treat anything involving unstable flags as experimental until it’s proven solid. Happy to help with that if there’s agreement.

**Comment by michielbdejong** [comment]
> The sarcastic and dismissive tone here — e.g. “let’s play logical fallacies again” and “failure to understand” — is not okay.

I totally agree, it's maybe not the only reason that we are, as a group, struggling to make peaceful progress in the Solid project, but sarcasm (especially in written comments) does not seem to be a useful tool for us, and it's probably doing more harm than good to the Solid project at this point.

@csarven from now on, please help to keep sarcasm out of written interactions if you can?

**Comment by csarven** [comment]
I am bowing out of this thread after this comment.

---

@michielbdejong , if you're genuinely keen on putting effort to making "peaceful progress", I'm so with you, but then I need to request that you ask yourself why you're rushing here attempting to tone police, focusing only on my comments and not others', and why you are not making any remarks on logical fallacies as pointed out, refraining from requesting citations on unsubstantiated claims, or calling out argumentum ad nauseam. Let me know when you'd like to have that rational discussion in good faith.

---

As for any remaining technical discussions on this issue, there has not been a strong argument put forward to revert anything. If anything, reverting is likely to cause more concerns - after all, 5.8 is out there already, which includes other fixes and features - than making efforts towards improving whatever needs to be done. Reverting is out of the question as I see it (or in any open source project in a similar situation). If anyone cares enough, and willing to put their time, they can focus on putting their energy on fixing or improving the broken bits. Alternatively, they can consider refraining from repeating their request (argumentum ad nauseam) - we understood it the first time.
